(Bug) Button + Spinner: Match Design

dminones opened the pull request:
<!-- reporting stories from Pivotal Tracker (replace 'x' by the Story's id) -->
This PR closes [#167839750](https://www.pivotaltracker.com/story/show/167839750), by:

<!-- If you're linking a repository issue, use the following line instead:
This PR closes #x, by:
-->

* Adding `ActivityIndicator` as loading component instead default Loading
* Updating `SaveButto` to make it work with new implementation
* Adding  `getDesignRelativeSize` that receives a width as parameter matching the design's width and converts to dp on the current device
* Adding tests to `getDesignRelativeSize`

**Extra:**

![Peek 2019-08-14 19-34](https://user-images.githubusercontent.com/1731297/63061491-ba9cfd80-becb-11e9-9304-4d3e66a994e7.gif)
![saveButton](https://user-images.githubusercontent.com/1731297/63061498-c25ca200-becb-11e9-9196-24b7b278f303.gif)

<!-- **Considerations:** -->
<!-- * PR Title Convention: (Feature|Bug|Chore) Task Name -->
<!-- * Be as descriptive as possible, assisting reviewers as much as possible. -->
<!-- * If frontend, paste screenshots that display the UI changes. -->
<!-- * If there is interactivity, paste a gif showing the feature. -->

dminones commented:
Changes requested done!
Save button is not exactly the same but is hard to make it fit into iPhone 5 devices so hoping that with these changes is good enough.

![nextbutton](https://user-images.githubusercontent.com/1731297/63102901-6426d200-bf52-11e9-8cc9-cf74ce9c6cad.gif)
![saveButtonProfile](https://user-images.githubusercontent.com/1731297/63102902-6426d200-bf52-11e9-8933-849168410bc4.gif)

LiavGut commented:
> * I think button with spinner should be in disabled mode - I agree
> * ask @LiavGut about changing the font size of the save button. - Don't change the font size (I don't want it to be unreadable), try to change the spinner size, let's see how it looks.

dminones commented:
Ok, now  `CustomButton` is disabled when loading. Also reduced a bit the spinner size and the disabled style also contributes. @LiavGut please confirm

![loadingDisabled](https://user-images.githubusercontent.com/1731297/63195238-75eaa100-c048-11e9-97d5-490687721100.gif)

LiavGut commented:
> Ok, now `CustomButton` is disabled when loading. Also reduced a bit the spinner size and the disabled style also contributes. @LiavGut please confirm
> 
> ![loadingDisabled](https://user-images.githubusercontent.com/1731297/63195238-75eaa100-c048-11e9-97d5-490687721100.gif)

Great, thanks.
Just a quick question, is the text aligned to the middle of the button hight? it looks like it needs to be a bit lower.

kbardi commented on src/components/common/buttons/CustomButton.js:
```diff
@@ -134,18 +162,22 @@ const CustomButton = (props: ButtonProps) => {
+        {icon && (!iconAlignment || iconAlignment === 'left') && (
+          <IconButton icon={icon} theme={theme} dark={dark} size={iconSize} style={styles.leftIcon} />
+        )}
+        {loading && <ActivityIndicator style={styles.activityIndicator} animating={loading} color={'#fff'} size={23} />}
```
I think Activity Indicator should be adapted to the font size...because in some cases we could need to change size...same with color if the button has a light color

dminones replied:
@kbardi  Changed color to use light or props.color depending on button mode.
Didn't add an activity indicator size since there is only one size right now and it's not clear if this value should be calculated from other prop or we should add a new prop.

sirpy commented:
@dariominones you didn't address the profile save button font size, has it been changed to fit? liav asked the font size shouldnt be changed

dminones commented:
> @dariominones you didn't address the profile save button font size, has it been changed to fit? liav asked the font size shouldn't be changed

@sirpy sorry, I misunderstood this. The font size was reduced to make it fit. However, I've made a few tries and this is how it looks with the same font size than other buttons.

![saveButtonBig](https://user-images.githubusercontent.com/1731297/63349373-a7f25080-c331-11e9-9a2b-78ebfcd7064b.gif)

@LiavGut what do you think?

sirpy commented:
> > @dariominones you didn't address the profile save button font size, has it been changed to fit? liav asked the font size shouldn't be changed
> 
> @sirpy sorry, I misunderstood this. The font size was reduced to make it fit. However, I've made a few tries and this is how it looks with the same font size than other buttons.
> 
> ![saveButtonBig](https://user-images.githubusercontent.com/1731297/63349373-a7f25080-c331-11e9-9a2b-78ebfcd7064b.gif)
> 
> @LiavGut what do you think?

@dariominones make spinner smaller and reduce paddings of button so it fits

dminones commented:
Changed avatar size to match with designs following our discussion with hadar. 
Changed to 14dp font size for save button according to designs

![iPhone6 gif](https://user-images.githubusercontent.com/1731297/63360937-9d41b680-c345-11e9-9335-db14c6ef65f0.gif)

![iPhone5](https://user-images.githubusercontent.com/1731297/63360939-9dda4d00-c345-11e9-8b77-5d24b8f93f86.gif)

sirpy commented on src/lib/utils/sizes.js:
```diff
@@ -0,0 +1,17 @@
+
+  const sizeInVW = width / DESIGN_WIDTH
+  const size = Math.min(width, screenWidth * sizeInVW)
+  console.info({ sizeInVW, size, screenWidth })
```
```suggestion
```

dminones commented:
> looks good.
> please refactor to getDesignRelativeWidth (width,isMax=true)
> if isMax=false then we don't do the Math.min
> please add also getDesignRelativeHeight

done

pablofullana commented:
@sirpy friendly ping

sirpy commented:
@dariominones resolve conflicts and squash and merge
